presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -360,20 +568,159 @@ protected Page computeNext()
                             BlockBuilder output = pageBuilder.getBlockBuilder(types.size() + i);
                             aggregator.evaluate(groupId, output);
                         }
-
-                        groupId++;
                     }
 
                     return pageBuilder.build();
                 }
             };
         }
+
+        private void rebuildAggregators()
+        {
+            groupByHash = createGroupByHash();
+            aggregators = createAggregators();
+        }
+
+        private void rebuildAggregatorsForMerge()
+        {
+            groupByHash = createGroupByHash();
+            aggregators = createAggregatorsForMerge(groupByHash.getTypes().size());
+        }
+
+        private GroupByHash createGroupByHash()
+        {
+            return GroupByHash.createGroupByHash(operatorContext.getSession(), groupByTypes, Ints.toArray(groupByChannels), hashChannel, expectedGroups);
+        }
+
+        private List<Aggregator> createAggregators()
```

Comment:
still from the name I cannot tell what the difference is. It would be nice if by looking at the two methods I could tell how they are different